File: block.py
from datetime import datetime
from config import GENESIS
from blockhash import blockhash


class Block:
    def __init__(self, timestamp: datetime, lasthash: str, hash: str, data) -> None:
        self.timestamp = timestamp
        self.lasthash = lasthash
        self.hash = hash
        self.data = data

    @classmethod
    def genesis(cls):
        return cls(**GENESIS)

    @classmethod
    def mine_block(cls, lastblock, data):
        block = cls(datetime.now(), lastblock.hash, None, data)
        block.hash = blockhash(block)
        return block

    def __str__(self) -> str:
        return f"{self.lasthash}{self.timestamp}{self.data}"

File: block.js
const { GENESIS_DATA } = require("./config");
const cryptoHash =require('./crypto-hash')

 class Block{  
    constructor({timestamp,lastHash,hash,data}){
        
        this.timestamp=timestamp;
        this.lastHash=lastHash;
        this.hash=hash;
        this.data=data;
    }
    static genesis(){
        return new Block(GENESIS_DATA);
    }
    static mineBlock({lastBlock, data}) {
        let block = new Block({
            timestamp: Date.now(),
            lastHash: lastBlock.hash,
            data
        });
        block.hash = cryptoHash(block.timestamp, block.lastHash, block.data);
        return block;
    }
   }
    
module.exports=Block;